lib/model/katex.dart

```diff
@@ -136,6 +133,123 @@ class _KatexParser {
   KatexNode _parseSpan(dom.Element element) {
     // TODO maybe check if the sequence of ancestors matter for spans.
 
+    if (element.className.startsWith('strut')) {
+      if (element.className == 'strut' && element.nodes.isEmpty) {
+        final styles = _parseSpanInlineStyles(element);
+        if (styles == null) throw KatexHtmlParseError();
+
+        final heightEm = styles.heightEm;
+        if (heightEm == null) throw KatexHtmlParseError();
+        final verticalAlignEm = styles.verticalAlignEm;
+
+        // Ensure only `height` and `vertical-align` inline styles are present.
+        if (styles.filter(heightEm: false, verticalAlignEm: false) !=
+            KatexSpanStyles()) {
```

Comment:
This will allocate two fresh KatexSpanStyles objects, with all their fields, in order to check that the other fields of `styles` are all null.

We can eliminate one of those allocations easily by saying `const KatexSpanStyles()`.

The other one is from `filter`. Let's leave that for now, I guess; it's probably not a big deal in practice.

(Definitely it's good to be doing this check, excluding unanticipated shapes of the tree.)